packages/component/src/Styles/CSSTokens.ts

```diff
@@ -0,0 +1,18 @@
+import CustomPropertyNames from './CustomPropertyNames';
+
+type CustomPropertyNamesType = typeof CustomPropertyNames;
+
+type CSSTokensType<T extends Readonly<Record<string, string>>> = {
```

Comment:
I can't use `Readonly<Record<keyof typeof CustomPropertyNames, string>>>`.

This is because the `var(${T[K]})` below... the `T[K]` will be evaluated as a non-string IDKW.

Reply:
I think this is the best way to do it.
```ts
const CustomPropertyNames = {
  foo: 'foo',
} as const;

type Tokens = typeof CustomPropertyNames;

type CustomPropertyNamesType = typeof CustomPropertyNames;

type CSSTokensType<T extends Readonly<Record<string, string>>> = {
  [K in keyof T]: `var(${T[K]})`;
};

// To add/remove/update a token, go to `CustomPropertyName.ts`.
const CSSTokens = new Proxy(
  {},
  {
    get(_, key: keyof CustomPropertyNamesType) {
      // We already checked in the `CustomPropertyName`.

      return `var(${CustomPropertyNames[key]})`;
    },
  }
) as CSSTokensType<Tokens>;
```
<img width="436" alt="image" src="https://github.com/microsoft/BotFramework-WebChat/assets/17039790/2ddcdb23-5841-486a-9698-fc01d7000503">